src/main/java/org/jenkinsci/maven/plugins/hpi/ValidateMojo.java

```diff
@@ -32,4 +41,42 @@ private static boolean isMustangOrAbove() {
             return false;
         }
     }
+
+    private void validateCoreVersion() throws MojoExecutionException {
+        try {
+            VersionNumber coreVersion = new VersionNumber(findJenkinsVersion());
+            MavenArtifact maxCoreVersionArtifact = null;
+            VersionNumber maxCoreVersion = new VersionNumber("0");
+
+            for (MavenArtifact artifact : Sets.union(getProjectArtfacts(), getDirectDependencyArtfacts())) {
```

Comment:
Not a blocker: Note that it does not take transitive dependencies into account. All plugins in the dependency tree will have to adopt the new Maven HPI Plugin version to prevent from errors in transitive deps. Related to #195 which also needs a full plugin dependency tree resolution.